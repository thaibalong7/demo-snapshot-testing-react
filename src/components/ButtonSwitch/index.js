/**
 *
 * ButtonSwitch
 *
 */

import React from 'react';
import PropTypes from 'prop-types';

function ButtonSwitch(props) {
  return (
    <div
      onClick={props.onChange}
      className={`button-switch ${props.checked ? 'active' : ''} ${
        props.disabled ? 'disabled' : ''
      }`}
    >
      <button
        type="button"
        className={`btn btn-sm btn-${props.checked ? 'success' : 'light'}`}
      />
      <label>{props.checked ? 'ON' : 'OFF'}</label>
    </div>
  );
}

ButtonSwitch.propTypes = {
  checked: PropTypes.bool,
  disabled: PropTypes.bool,
  onChange: PropTypes.func,
};

export default ButtonSwitch;
