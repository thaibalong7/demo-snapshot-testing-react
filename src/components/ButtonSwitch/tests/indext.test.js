/**
 * Testing ButtonSwitch component
 *
 * Run `jest app/components/core/buttonswitch` to test this
 * Run `jest app/components/core/buttonswitch -u` to update file snapshot
 *
 */
import React from 'react';
import renderer from 'react-test-renderer';

import ButtonSwitch from '../index';


describe('Testing ButtonSwitch', () => {
  let component;
  const mockOnChange = jest.fn();

  /**
   * Runs a function after each one of the tests in this file completes.
   * If the function returns a promise or is a generator, Jest waits for that promise to resolve before continuing.
   */
  afterEach(() => {
    mockOnChange.mockClear()
  });

  it('renders correctly - [checked-false] [disabled-false]', () => {
    component = renderer
      .create(<ButtonSwitch
        checked={false}
        disabled={false}
        onChange={mockOnChange}
      />);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly - [checked-true] [disabled-false]', () => {
    component = renderer
      .create(<ButtonSwitch
        checked
        disabled={false}
        onChange={mockOnChange}
      />);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly - [checked-true] [disabled-true]', () => {
    component = renderer
      .create(<ButtonSwitch
        checked
        disabled
        onChange={mockOnChange}
      />);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly - [checked-false] [disabled-true]', () => {
    component = renderer
      .create(<ButtonSwitch
        checked={false}
        disabled
        onChange={mockOnChange}
      />);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should call props.onChange when click switch button with `disabled` is `false`', () => {
    component = renderer
      .create(<ButtonSwitch
        checked={false}
        disabled={false}
        onChange={mockOnChange}
      />);

    const tree = component.toJSON();
    tree.props.onClick();

    // props onChange should called once time
    expect(mockOnChange.mock.calls.length).toBe(1);
  });

});
