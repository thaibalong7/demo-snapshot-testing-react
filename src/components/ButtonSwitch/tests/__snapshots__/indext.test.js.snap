// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ButtonSwitch renders correctly - [checked-false] [disabled-false] 1`] = `
<div
  className="button-switch  "
  onClick={[MockFunction]}
>
  <button
    className="btn btn-sm btn-light"
    type="button"
  />
  <label>
    OFF
  </label>
</div>
`;

exports[`Testing ButtonSwitch renders correctly - [checked-false] [disabled-true] 1`] = `
<div
  className="button-switch  disabled"
  onClick={[MockFunction]}
>
  <button
    className="btn btn-sm btn-light"
    type="button"
  />
  <label>
    OFF
  </label>
</div>
`;

exports[`Testing ButtonSwitch renders correctly - [checked-true] [disabled-false] 1`] = `
<div
  className="button-switch active "
  onClick={[MockFunction]}
>
  <button
    className="btn btn-sm btn-success"
    type="button"
  />
  <label>
    ON
  </label>
</div>
`;

exports[`Testing ButtonSwitch renders correctly - [checked-true] [disabled-true] 1`] = `
<div
  className="button-switch active disabled"
  onClick={[MockFunction]}
>
  <button
    className="btn btn-sm btn-success"
    type="button"
  />
  <label>
    ON
  </label>
</div>
`;
